### PR TITLE
mobile: Set the socket receive buffer size to 1MB for QUIC

### PR DESCRIPTION
**Pull request by abeyad**
Cronet sets the UDP receive buffer size to 1MB:
https://source.chromium.org/chromium/chromium/src/+/main:net/quic/quic_session_pool.cc;l=839;drc=7dc3003765c74daa4bb49e4463e3a7963d8c7ea1

If we don't set the receive buffer size explicitely, the operating system defaults are used, which are typically too small for UDP.

### CODE REVIEW

**Review by alyssawilk — DISMISSED** [review]
ideally we'd have a TODO to clean up once Enovy had solid defaults but I trust you on it :-P

**Comment by abeyad** [comment]
> ideally we'd have a TODO to clean up once Enovy had solid defaults but I trust you on it :-P

Added a TODO with the issue ID

**Review by alyssawilk — APPROVED** [review]

File: mobile/library/cc/engine_builder.cc [hunk 1]
```diff
@@ -754,6 +766,20 @@ std::unique_ptr<envoy::config::bootstrap::v3::Bootstrap> EngineBuilder::generate
     (*base_cluster->mutable_typed_extension_protocol_options())
         ["envoy.extensions.upstreams.http.v3.HttpProtocolOptions"]
             .PackFrom(alpn_options);
+
+    // Set the upstream connections socket receive buffer size. The operating system defaults are
+    // usually too small for QUIC.
+    // TODO(32304): We can remove this once core Envoy has better receive buffer defaults.
```
Comment (danzh2010) [inline mobile/library/cc/engine_builder.cc:772]: I think we should explicitly set this in EM regardless of the default value in core Envoy. We know 1MB is sufficient for mobile but not necessarily for server upstream connections.

Comment (alyssawilk) [reply mobile/library/cc/engine_builder.cc:772]: fair point!

Comment (abeyad) [reply mobile/library/cc/engine_builder.cc:772]: oops sorry just saw this.  do we plan on making the core Envoy change configured via `bind_options` just like we do here? in that case, we probably don't need this TODO, and I'll send a follow up PR to remove it

Comment (abeyad) [reply mobile/library/cc/engine_builder.cc:772]: https://github.com/envoyproxy/envoy/pull/32337

**Pull request merged**